jest.config.js

```diff
@@ -9,16 +31,14 @@ const tsJestOptions = {
 	},
 };
 
-const { baseUrl, paths } = require('get-tsconfig').getTsconfig().config?.compilerOptions;
-
 /** @type {import('jest').Config} */
 const config = {
 	verbose: true,
 	testEnvironment: 'node',
 	testRegex: '\\.(test|spec)\\.(js|ts)$',
 	testPathIgnorePatterns: ['/dist/', '/node_modules/'],
 	transform: {
-		'^.+\\.ts$': ['ts-jest', tsJestOptions],
+		'^.+\\.ts$': useSwc ? ['@swc/jest', swcJestOptions] : ['ts-jest', tsJestOptions],
```

Comment:
1. For some packages we'll be using `tsc` for `build` and `dev` vs. swc for `test`. Curious if there is a risk that the dist outputs might differ, causing us to miss issues?
2. `cli` is where the bulk of BE tests are and the [linked issue](https://github.com/swc-project/swc/issues/6824) says swc is unlikely to support emitting decorator metadata soon. This really limits the upside of this change.

```
core @swc/jest 1.59s
core ts-jest 1.85s

nodes-base @swc/jest 23.978s
nodes-base ts-jest 33.549s

workflow @swc/jest 3.813s
workflow ts-jest 5.33s
```

Wondering if you've tried `esbuild-jest` and found it limited as well? For reference `bun test` doesn't yet support all the Jest APIs we're using, e.g. `jest.restoreAllMocks()`.